src/Microsoft.OData.Core/ODataResource.cs

```diff
@@ -391,5 +396,27 @@ public void AddFunction(ODataFunction function)
                 this.functions.Add(function);
             }
         }
+
+        private static void VerifyProperties(IEnumerable<ODataProperty> properties)
+        {
+            if (properties != null)
+            {
+                ODataCollectionValue collection = null;
+                foreach (var property in properties)
+                {
+                    if (property.Value is ODataResourceValue)
+                    {
+                        throw new ODataException(Strings.ODataResource_PropertyValueCannotBeODataResourceValue(property.Name));
+                    }
+                    else if ((collection = property.Value as ODataCollectionValue) != null)
+                    {
+                        if (collection.Items != null && collection.Items != null && collection.Items.Any(t => t is ODataResourceValue))

```

Comment:
>collection.Items != null [](start = 56, length = 24)

You check this twice -- you really don't want collection.Items to be null! :-)

Reply:
copied and pasted in the wrong place. Thanks for catching